### ISSUE TITLE
modal: improve focus behaviour

### ISSUE THREAD

**Issue by aristov** [bem/bem-components#1674]
Here are the recommendations from the 3rd point of [Understanding WCAG: Focus order](http://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-order.html#navigation-mechanisms-focus-order-examples-head) for modal dialogs:
- [x] ~~When the trigger button is activated, a dialog opens~~
- [ ] and focus is set to the first interactive element in the dialog
- [ ] As long as the dialog is open, focus is limited to the elements of the dialog.
- [ ] When the dialog is dismissed, focus returns to the button or the element following the button.

`modal` satisfies the crossed out items. We should implement the rest.

**Comment by aristov** [comment]
The issue description was merged to #1673 as `modal` is implemented via `popup` and these blocks should be fixed together.

**Comment by tadatuta** [comment]
Modal with focus management solved https://github.com/edenspiekermann/accessible-modal-dialog
